default/AI/ResearchAI.py

```diff
@@ -379,53 +394,141 @@ def generate_research_orders():
-            tuple(AIDependencies.DAMAGE_CONTROL_TECHS): partial(if_enemies, 0.1, 0.5),
-            tuple(AIDependencies.HULL_TECHS): get_hull_priority,
-            tuple(AIDependencies.ARMOR_TECHS): partial(ship_usefulness, if_enemies, MIL_IDX),
-            tuple(AIDependencies.ENGINE_TECHS): partial(ship_usefulness, partial(if_dict, choices, 'engine', true_val=0.6), None),
-            tuple(AIDependencies.FUEL_TECHS): partial(ship_usefulness, partial(if_dict, choices, 'fuel'), None),
-            tuple(AIDependencies.SHIELD_TECHS): partial(ship_usefulness, if_enemies, MIL_IDX),
-            tuple(AIDependencies.TROOP_POD_TECHS): partial(ship_usefulness, partial(if_enemies, 0.1, 0.3), TROOP_IDX),
-            tuple(AIDependencies.COLONY_POD_TECHS): partial(ship_usefulness, partial(const_priority, 0.5), COLONY_IDX),
-        })
-
-    # keys are individual full tech names
-    if not individual_priority_funcs:
-        individual_priority_funcs.update({
-            AIDependencies.PRO_MICROGRAV_MAN: partial(conditional_priority, partial(const_priority, 3.5), priority_low, None, ColonisationAI, 'got_ast'),
-            AIDependencies.PRO_ORBITAL_GEN: partial(conditional_priority, partial(const_priority, 3.0), priority_low, None, ColonisationAI, 'got_gg'),
-            AIDependencies.PRO_SINGULAR_GEN: partial(conditional_priority, partial(const_priority, 3.0), priority_low, partial(has_star, fo.starType.blackHole)),
-            AIDependencies.GRO_XENO_GENETICS: get_xeno_genetics_priority,
-            AIDependencies.LRN_XENOARCH: priority_low if foAI.foAIstate.aggression < fo.aggression.typical else partial(conditional_priority, partial(const_priority, 5.0), priority_low, None, ColonisationAI, 'gotRuins'),
-            AIDependencies.LRN_ART_BLACK_HOLE: get_artificial_black_hole_priority,
-            AIDependencies.GRO_GENOME_BANK: priority_low,
-            AIDependencies.CON_CONC_CAMP: partial(priority_zero),
-            AIDependencies.NEST_DOMESTICATION_TECH: priority_zero if foAI.foAIstate.aggression < fo.aggression.typical else partial(conditional_priority, partial(const_priority, 3.0), priority_low, None, ColonisationAI, 'got_nest'),
-        })
-
-    if not ALL_MISC_GROUP_TECH_NAMES:
-        ALL_MISC_GROUP_TECH_NAMES.extend(list(tech_name for misc_group in misc_group_priority_funcs for tech_name in misc_group))
-
-
-
-    report_adjustments = False
+    if not priority_funcs:
+        tech_handlers = (
+            (
+                AIDependencies.PRO_MICROGRAV_MAN,
+                partial(conditional_priority, partial(const_priority, 3.5), priority_low, None, ColonisationAI,
+                        'got_ast')
+            ),
+            (
+                AIDependencies.PRO_ORBITAL_GEN,
+                partial(conditional_priority, partial(const_priority, 3.0), priority_low, None, ColonisationAI,
+                        'got_gg')
+            ),
+            (
```

Comment:
syntax for multiline imports.
```python
from from AIDependencies import (PRO_MICROGRAV_MAN,
                                                         PRO_ORBITAL_GEN)
```

`Dep` will be better because you can use autocomplete with out remembering exact names)